jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/CreateQueueDialog.kt

```diff
@@ -64,42 +66,44 @@ class CreateQueueDialog(
     }
 
     private fun validateFields(): ValidationInfo? {
-        if (queueName().isEmpty()) {
+        if (view.queueName.text.isEmpty()) {
             return ValidationInfo(message("sqs.create.validation.empty.queue.name"), view.queueName)
-        } else if (queueName().length > MAX_LENGTH_OF_QUEUE_NAME) {
-            return ValidationInfo(message("sqs.create.validation.long.queue.name", MAX_LENGTH_OF_QUEUE_NAME), view.queueName)
         }
-
-        if (view.fifoType.isSelected) {
-            if (!queueName().endsWith(".fifo")) {
-                return ValidationInfo(message("sqs.create.validation.fifo"), view.queueName)
-            } else if (!validateCharacters(queueName().substringBefore(".fifo"))) {
-                return ValidationInfo(message("sqs.create.validation.queue.name.invalid"), view.queueName)
-            }
-        } else {
-            if (!validateCharacters(queueName())) {
-                return ValidationInfo(message("sqs.create.validation.queue.name.invalid"), view.queueName)
+        if (!validateCharacters(view.queueName.text)) {
+            return ValidationInfo(message("sqs.create.validation.queue.name.invalid"), view.queueName)
+        }
+        if (queueName().length > MAX_LENGTH_OF_STANDARD_QUEUE_NAME) {
+            return if (view.fifoType.isSelected) {
+                ValidationInfo(message("sqs.create.validation.long.queue.name", MAX_LENGTH_OF_FIFO_QUEUE_NAME), view.queueName)
+            } else {
+                ValidationInfo(message("sqs.create.validation.long.queue.name", MAX_LENGTH_OF_STANDARD_QUEUE_NAME), view.queueName)
             }
         }
 
         return null
     }
 
-    private fun validateCharacters(queueName: String): Boolean = queueName.matches("^[-a-zA-Z0-9_]*$".toRegex())
+    private fun validateCharacters(queueName: String): Boolean = queueName.matches("^[a-zA-Z0-9-_]*$".toRegex())
 
-    private fun queueName() = view.queueName.text.trim()
+    private fun queueName(): String {
+        var name = view.queueName.text.trim()
+        if (view.fifoType.isSelected) {
```

Comment:
(not sure if we should actually do this) but we can not append .fifo if the user already did (like if they missed the box), otherwise the failure message would be kind of confusing.

Reply:
I can see how that could work for users who are used to web console, but it might be a little confusing with receiving different length errors based on input. Also, now it might be confusing in the case that the user misspells '.fifo' and receives a failure error about invalid characters.